src/LibraryManager/CacheService/WebRequestHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager
+{
+    /// <summary>
+    /// Helper class to hold the HttpClient instance and send requests to get resources
+    /// </summary>
+    internal class WebRequestHandler : IWebRequestHandler, IDisposable
+    {
+        private HttpClient _httpClient;
+        private WebRequestHandler()
+        {
+            HttpClientHandler httpMessageHandler = new HttpClientHandler();
+            httpMessageHandler.MaxConnectionsPerServer = 1000;
+
+            _httpClient = new HttpClient(httpMessageHandler);
+        }
+
+        public static IWebRequestHandler Instance { get; } = new WebRequestHandler();
+
+        public void Dispose()
+        {
+            _httpClient.Dispose();
+        }
+
+        public async Task<Stream> GetStreamAsync(string url, CancellationToken cancellationToken)
+        {
+            try
+            {
+                return await _httpClient.GetStreamAsync(url).WithCancellation(cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                throw new ResourceDownloadException(url);
```

Comment:
Set the caught exception as InnerException?

Reply:
Do we need this info?

Reply:
We need to log telemetry on that

Reply:
Is this the right place to log that telemetry?  This seems like just a utility and can't provide good perspective on the operation that failed?

Reply:
Not at this file, but wherever we are treating this exception.